docs: update HR with indicator (#17435)
unoplatform/uno: doc/articles/features/working-with-xaml-hot-reload.md

> - XAML and [C# Markup](xref:Uno.Extensions.Markup.Overview) Hot Reload for **iOS, Catalyst, Android, WebAssembly, and Skia (X11, Windows, macOS and FrameBuffer)**

Reviewer: @dr1rrb / @rajamatt / @jeromelaban 
Outside the addition of the documentation for the Hot Reload Indicator, do we have additional updates needed for this file for Hot Reload in general, please?

- https://platform.uno/docs/articles/features/working-with-xaml-hot-reload.html
- https://platform.uno/docs/articles/uno-development/uno-internals-hotreload.html

> - All **[C# of Hot Reload](https://learn.microsoft.com/visualstudio/debugger/hot-reload)** in both Visual Studio and VS Code. See [supported code changes](https://learn.microsoft.com/visualstudio/debugger/supported-code-changes-csharp).

Reviewer: @rajamatt you will need to fix some spell check and markdown linting issues:
![image](https://github.com/unoplatform/uno/assets/16295702/fa2ccef8-60ca-40b5-b332-528eeaa3c855)
- [Markdown linting errors](https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=132025&view=logs&jobId=e61e0ee1-f3da-5962-52cf-48bdde3a5bd4&j=e61e0ee1-f3da-5962-52cf-48bdde3a5bd4&t=c18ecb7c-3c97-55c3-9bb9-5ef1777cbc25)
- [Spell check errors](https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=132025&view=logs&jobId=050fb64b-9b12-52cc-b750-ec5d4e778b08&j=050fb64b-9b12-52cc-b750-ec5d4e778b08&t=669790bd-240a-5aea-d705-86a7326c36e7)

*@rajamatt I will suggest you to install the markdown lint extension for VS Code, helpful while writing documentation to test locally for markdown linting errors
https://marketplace.visualstudio.com/items?itemName=DavidAnson.vscode-markdownlint

- **Simulator and physical devices** support

> - Hot Reload **Indicator** visuals for an easier development experience

Reviewer 2: ```suggestion
- Hot Reload **Indicator** visuals for an enhanced development experience
```

> - Hot Reload **Indicator** visuals for an enhanced development experience

Reviewer: ```suggestion
- Hot Reload **Indicator** visuals for an enhanced development experience (Except WinAppSDK)
```
With a mention like this ?

Reviewer 2: Maybe something like:
```suggestion
- Hot Reload **Indicator** visuals for an enhanced development experience on Uno Platform targets
```

Hot Reload is supported by Visual Studio for WinAppSDK and provides support in unpackaged deployment mode.

> ## Hot Reload Indicator

Reviewer: @rajamatt, @dr1rrb, @jeromelaban 
I think we should mention that the Hot Reload Indicator is not available for WinAppSDK Packaged/Unpackaged

Reviewer 2: Agreed.

> Hot Reload also has a visual indicator to help you further monitor changes while developing. It displays new information every time Hot Reload is triggered. The indicator is enabled by default within the `EnableHotReload()` method which is located in the root App.xaml.cs file. This displays an overlay which hosts the indicator. If you wish to disable it, you simply have to provide the following boolean: `EnableHotReload(disableIndicator: true)`, removing the overlay from the view.

Reviewer 2: ```suggestion
Hot Reload displays a visual indicator to help you further monitor changes while developing. It displays new information every time Hot Reload is triggered. The indicator is enabled by default within the `EnableHotReload()` method which is located in the root `App.xaml.cs` file. This displays an overlay which hosts the visual indicator. If you wish to disable it, you simply have to provide the following boolean: `EnableHotReload(disableIndicator: true)`, removing the overlay from the view.
```

> Hot Reload displays a visual indicator to help you further monitor changes while developing. It displays new information every time Hot Reload is triggered. The indicator is enabled by default within the `EnableHotReload()` method which is located in the root `App.xaml.cs` file. This displays an overlay which hosts the visual indicator. If you wish to disable it, you simply have to provide the following boolean: `EnableHotReload(disableIndicator: true)`, removing the overlay from the view.

Reviewer: ```suggestion
Hot Reload displays a visual indicator to help you further monitor changes while developing. It displays new information every time Hot Reload is triggered. The indicator is enabled by default within the `EnableHotReload()` method which is located in the root `App.xaml.cs` file. This displays an overlay that hosts the visual indicator. If you wish to disable it, you simply have to provide the following boolean: `EnableHotReload(disableIndicator: true)`, removing the overlay from the view.
```

<p align="center">

>   <img src="../Assets/features/hotreload/indicator.png" alt="HotReloadIndicator" />

Reviewer 2: ```suggestion
  <img src="../Assets/features/hotreload/indicator.png" alt="A hot reload visual indicator" />
```

</p>

> [!TIP]

> > The overlay can be moved by using the anchor on the left hand side.

Reviewer: ```suggestion
> The overlay can be moved by using the anchor on the left-hand side.
```

> > The overlay is grabbable by the anchor on the left hand side and can be moved around so that it doesn't hide the view you are modifying.

Reviewer 2: ```suggestion
> The overlay can be moved by using the anchor on the left hand side.
```

> The indicator displays the current connection status. Clicking on it will open a flyout containing all events or changes that were made recorded by Hot Reload. These events display more details about Hot Reload changes, like its status and impacted files. Connection event statuses are also shown.

Reviewer 2: ```suggestion
The indicator displays the current connection status. Clicking on it will open a flyout containing all events or changes that were applied by Hot Reload. These events display more details about Hot Reload changes, such as its status and impacted files.
```

<p align="center">

>   <img src="../Assets/features/hotreload/indicator-flyout.png" alt="HotReloadIndicatorFlyout" />

Reviewer 2: ```suggestion
  <img src="../Assets/features/hotreload/indicator-flyout.png" alt="An window showing events from Hot Reload" />
```

</p>

### Statuses

Here's a summary of what icons and statuses you can expect:

__Connection__:

> ![HotReloadIconConnectionSuccess](../Assets/features/hotreload/status-connection-success.png) Success

Reviewer 2: ```suggestion
![The icon shown on a successful connection](../Assets/features/hotreload/status-connection-success.png) Success
```

> ![HotReloadIconConnectionFailed](../Assets/features/hotreload/status-connection-failed.png) Failed (Hot Reload was not able to connect to the endpoint or lost connection)

Reviewer 2: ```suggestion
![The icon shown on a failed connection](../Assets/features/hotreload/status-connection-failed.png) Failed (Hot Reload was not able to connect to the endpoint or lost connection)
```

> ![HotReloadIconConnectionWarning](../Assets/features/hotreload/status-connection-warning.png) Warning (Version mismatch, unknown updates or processors missing)

Reviewer 2: ```suggestion
![The icon shown on a successful connection, with warnings](../Assets/features/hotreload/status-connection-warning.png) Warning (Version mismatch, unknown updates or processors missing)
```

> ![HotReloadIconConnectionWarning](../Assets/features/hotreload/status-connection-warning.png) Warning (Version mismatch, unknown updates or processors missing)

Reviewer 3: Wondering if we should repeat here content of messages to be more actionable. Maybe a simple message "this usually indicates an issue that can be solved by restarting your IDE".

Author: Good point, the icons speak for themselves in most cases, its better to have text that could help the user

__Operation__:

> ![HotReloadIconSuccess](../Assets/features/hotreload/status-success.png) Success

Reviewer 2: ```suggestion
![The icon shown when Hot Reload succeeded](../Assets/features/hotreload/status-success.png) Success
```

> ![HotReloadIconFailed](../Assets/features/hotreload/status-failed.png) Failed (Rude edits, compilation errors and general errors)

Reviewer 2: ```suggestion
![The icon shown when Hot Reload failed](../Assets/features/hotreload/status-failed.png) Failed (Rude edits, compilation errors and general errors)
```

> ![HotReloadIconProcessing](../Assets/features/hotreload/status-processing.png) Processing (Hot Reload is processing the changes or initializing)

Reviewer 2: ```suggestion
![The icon shown when Hot Reload is in progress](../Assets/features/hotreload/status-processing.png) Processing (Hot Reload is processing the changes or initializing)
```
